src/lib/form-field/form-field.ts

```diff
@@ -144,12 +144,14 @@ export class MatFormField extends _MatFormFieldMixinBase
 
   /** The form-field appearance style. */
   @Input()
-  get appearance(): MatFormFieldAppearance {
-    return this._appearance || this._defaultOptions && this._defaultOptions.appearance || 'legacy';
-  }
+  get appearance(): MatFormFieldAppearance { return this._appearance; }
   set appearance(value: MatFormFieldAppearance) {
     const oldValue = this._appearance;
-    this._appearance = value;
+
+    if (value) {
```

Comment:
I think previously if you passed null it was the same as setting it to legacy. This seems to be changing it to ignore null, is there a reason for that?

Reply:
Maybe we can do `this._appearance = this._defaults && this._defaults.appearance || 'legacy'` instead?

Reply:
The reason for adding the null check was to avoid having to repeat `this._appearance = this._defaults && this._defaults.appearance || 'legacy'` in two places.

Reply:
yeah but people can still set `null` via a template binding

Reply:
Done.